### ISSUE TITLE
SecurityPolicyViolationEventInit data members naming doesn't seem to match browsers

### ISSUE THREAD

**Issue by cdumez** [w3c/webappsec-csp#498]
SecurityPolicyViolationEventInit data members naming doesn't seem to match browsers. In particular:
- WebKit & Blink use documentURI & blockedURI while the specification uses documentURL & blockedURL
- WebKit & Blink use lineNumber & columnNumber while the specification uses lineno & colno
- WebKit & Blink have violatedDirective as a required data member but it is not present in the spec

The SecurityPolicyViolationEvent interface itself is fine because it has attributes with both names so it is backward compatible. However, the SecurityPolicyViolationEventInit seems to be an issue.

In my point of view (I work on WebKit), it seems risky compatibility-size for browser engines to change the naming of our SecurityPolicyViolationEventInit dictionary data members. It is also not really feature-detectable on JS side as far as I can tell.

**Comment by cdumez** [comment]
WebKit bug where we got closer to the specification but did not rename the dictionary members out of fear of breaking compatibility: https://bugs.webkit.org/show_bug.cgi?id=226215

**Comment by darinadler** [comment]
I think you also mentioned that WPT currently has tests that match the browsers, not the specification.

**Comment by gsnedders** [comment]
c.f. https://github.com/w3c/webappsec-csp/pull/353

**Comment by cdumez** [comment]
> I think you also mentioned that WPT currently has tests that match the browsers, not the specification.

That is correct:
- content-security-policy/securitypolicyviolation/constructor-required-fields.html
Uses documentURI / blockedURI / lineNumber / columnNumber for dictionary members
Also expects violatedDirective as a mandatory parameter

- interfaces/CSP.idl matches the spec naming for dictionary members but those don't cause failures WPT test failures currently. In particular, content-security-policy/securitypolicyviolation/idlharness.window.html passes with old or new dictionary member naming.

**Comment by cdumez** [comment]
cc @mikewest @annevk

**Comment by cdumez** [comment]
I have just verified that Gecko is also still using the documentURI / blockedURI / lineNumber / columnNumber naming. So it seems that none of the 3 major browser engines match the specification, even though the specification changed the naming years ago.

**Comment by mikewest** [comment]
That's unfortunate, and likely my fault (at least for Blink).

At this point, I agree that shifting the spec to match browsers is the right thing to do. @antosart might have time to put up a PR?

**Comment by antosart** [comment]
I will prepare a PR.

**Comment by cdumez** [comment]
Does this mean also dropping the documentURL / blockedURL attributes on SecurityPolicyViolationEvent interface? It seems no browser has adopted these either. While they don't hurt compatibility, they don't bring much and won't match the properties on the dictionary anyway.

**Comment by annevk** [comment]
Yeah, we should revert all of #353 for now and then only reapply if there's more agreement on shipping (and writing test updates).
